### PR TITLE
[Mime] Add DataPart::setContentId()

### LINKED ISSUE #46959: Add support for setting custom CID for image attachment
### Description

In SwiftMailer this was supported, and I was able to set custom CID for inline image attachments.

The reason, why I need this is, that I have prepared templates for an email with CID already in place. However, attachment is generated and added on the fly. This is what I have in my HTML template:

`QR Payment`

### Example

My current solution includes custom DataPart class, which overrides the Symfony one. However, I think this one could be implemented directly in DataPart class.

```php
$image = new DataPart($qrCode->toPngText(), 'QR Payment.png', 'image/png');
$image->setContentId('qrpayment@sunapp.generated');
$symfonyMessage->attachPart($image);
```

```php
<?php

declare(strict_types=1);

namespace SunApp\Mail\Symfony;

use Symfony\Component\Mime\Header\Headers;
use Symfony\Component\Mime\Part\DataPart as SymfonyDataPart;

class DataPart extends SymfonyDataPart
{
    private ?string $contentId;

    public function setContentId(string $cid): void
    {
        $this->contentId = $cid;
    }

    public function getContentId(): string
    {
        return $this->contentId ?: $this->contentId = $this->generateMyContentId();
    }

    public function hasContentId(): bool
    {
        return null !== $this->contentId;
    }

    public function getPreparedHeaders(): Headers
    {
        $headers = parent::getPreparedHeaders();

        if (null !== $this->contentId) {
            $headers->setHeaderBody('Id', 'Content-ID', $this->contentId);
        }

        if (null !== $this->getFilename()) {
            $headers->setHeaderParameter('Content-Disposition', 'filename', $this->getFilename());
        }

        return $headers;
    }

    private function generateMyContentId(): string
    {
        return \bin2hex(\random_bytes(16)) . '@sunapp.generated';
    }
}

```

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | Fix #46959 
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Symfony/Component/Mime/Part/DataPart.php [hunk 1]
```diff
@@ -99,6 +99,11 @@ public function hasContentId(): bool
         return null !== $this->cid;
     }
 
+    public function setContentId(string $cid): void
```
Comment (arxeiss) [inline src/Symfony/Component/Mime/Part/DataPart.php]: Thank you for taking over my Issue and creating PR. However, I'm not sure this is enough.
As per http://www.faqs.org/rfcs/rfc2392.html CID should be in URL format.

So I'm not sure if this method shouldn't throw an exception when cid is invalid. I think, that old SwiftMailer was complaining about it. And also this codes generates CID in something, what looks like URL

```php
private function generateContentId(): string
{
    return bin2hex(random_bytes(16)).'@symfony';
}
```

Comment (fabpot) [reply src/Symfony/Component/Mime/Part/DataPart.php]: Maybe we can check that there is at least `@` in the id and throw an exception if not? @arxeiss Would you like to submit a PR for that?

Comment (arxeiss) [reply src/Symfony/Component/Mime/Part/DataPart.php]: OK, I can. Is that OK to throw just https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Mime/Exception/InvalidArgumentException.php? Or do I need to create some specific one?

Comment (fabpot) [reply src/Symfony/Component/Mime/Part/DataPart.php]: Seems like the right exception to throw indeed.

**Pull request merged**